js/ace.js

```diff
@@ -0,0 +1,1046 @@
+            'cost': undefined,
+            'average': average,
+            'filled': filled,
+            'remaining': remaining,
+            'status': status,
+            'fee': undefined,
+            'trades': undefined,
+            'info': undefined,
+        }, market);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        /**
+         * @method
+         * @name ace#createOrder
+         * @description create a trade order
+         * @see https://github.com/ace-exchange/ace-official-api-docs/blob/master/api_v2.md#open-api---new-order
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float|undefined} price the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const orderType = type.toUpperCase ();
+        const orderSide = side.toUpperCase ();
+        const request = {
+            'baseCurrencyId': market['baseId'],
+            'quoteCurrencyId': market['quoteId'],
+            'type': (orderType === 'LIMIT') ? 1 : 2,
+            'buyOrSell': (orderSide === 'BUY') ? 1 : 2,
+            'num': this.amountToPrecision (symbol, amount),
+        };
+        if (type === 'limit') {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        const response = await this.privatePostV2OrderOrder (this.extend (request, params), params);
+        //
+        //     {
+        //         "attachment": "15697850529570392100421100482693",
+        //         "message": null,
+        //         "parameters": null,
+        //         "status": 200
+        //     }
+        //
+        const data = this.safeValue (response, 'attachment');
+        return this.parseOrder (data, market);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        /**
+         * @method
+         * @name ace#cancelOrder
+         * @description cancels an open order
+         * @see https://github.com/ace-exchange/ace-official-api-docs/blob/master/api_v2.md#open-api---cancel-order
+         * @param {string} id order id
+         * @param {string} symbol unified symbol of the market the order was made in
+         * @param {object} params extra parameters specific to the ace api endpoint
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
```

Comment:
for some reason `cancelOrder` returns this error, can you check it?
```
Node.js: v17.9.0
CCXT v2.6.47
ace.cancelOrder (16745534865073930062681100739118)
BadRequest ace {"attachment":null,"message":"委託單已成交或已撤銷","parameters":null,"status":2061}
---------------------------------------------------
[BadRequest] ace {"attachment":null,"message":"委託單已成交或已撤銷","parameters":null,"status":2061}

    at throwExactlyMatchedException  js/base/Exchange.js:2419              throw new exact[string] (message);                                              
    at handleErrors                  js/ace.js:1042                        this.throwExactlyMatchedException (this.exceptions['exact'], status, feedback); 
    at                               js/base/Exchange.js:603               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                  
    at async timeout                 js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                        js/base/Exchange.js:2042              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                       js/base/Exchange.js:2046              return await this.fetch2 (path, api, method, params, headers, body, config, con…
    at cancelOrder                   js/ace.js:632                         const response = await this.privatePostV2OrderCancel (this.extend (request, par…
    at async run                     examples/js/cli.js:293                const result = await exchange[methodName] (... args)                            

ace {"attachment":null,"message":"委託單已成交或已撤銷","parameters":null,"status":2061}
```

Reply:
@carlosmiei it seems the order was settled, I can find the trades. 

They didn't return `orderType` in `fetchOrder` api, not sure whether it was market order or not.

Reply:
> @carlosmiei it seems the order was settled, I can find the trades.
> 
> They didn't return `orderType` in `fetchOrder` api, not sure whether it was market order or not.
> 
> I've submitted a `USDT/TWD` order, you can cancel that order (should work).

@sc0Vu I will re-check it thanks